File: React_Native/React_Native/src/pages/sat_pages/ferramentasSat.js
import React, { useState } from 'react';
import {View, Text, StyleSheet, TouchableOpacity, TextInput, Alert, DeviceEventEmitter} from 'react-native';

import OperacaoSat from '../../services/operacaoSat';
import RetornoSat from '../../configs/config_sat/retornoSat';

import ValidaCodigo from '../../components/ValidacaoCodigo';

const ferramentasSat = () => {
    const [ codigoAtivacao, setCodigoAtivacao ] = useState(global.MyVar);

    function dialogoSat(messageText){
        Alert.alert("Retorno", messageText);
    }

    function ferramentasSat(funcao){
        // Salva  o código de ativação para o usuario não precisar ficar digitando em todas as telas 
        global.MyVar = codigoAtivacao;

        var validationCodigo = new ValidaCodigo();

        if(validationCodigo.getValidation(codigoAtivacao)){
            //ENVIA PARA OPERAÇÃO SAT ONDE VAI CONECTAR COM O SAT DE ACORDO COM A FUNÇÃO
            var operacaoSat = new OperacaoSat();

            const args = {
                'funcao': funcao, 
                'random': Math.floor(Math.random() * 9999999).toString(),
                'codigoAtivar': codigoAtivacao
            };

            operacaoSat.invocarOperacaoSat(args);
            
            DeviceEventEmitter.once("eventFerramenta", event => { 
                var retorno = event.ferramenta;
                var retornoSat = new RetornoSat(args['funcao'], retorno);
                dialogoSat(operacaoSat.formataRetornoSat(retornoSat));
            });
        }else{
            dialogoSat("Código de Ativação deve ter entre 8 a 32 caracteres!");
        }
     }

    return (
        <View style={styles.container}>
            <View style={styles.containerOptionsAndButtons}>
                <Text style={styles.textOptions}>Código de Ativação Sat: </Text>
                <TextInput
                    style={styles.inputText}
                    autoCapitalize="none"
                    autoCorrect={false}
                    value={codigoAtivacao}
                    onChangeText={setCodigoAtivacao}
                />

                <View style={styles.containerButtons}>                
                    <TouchableOpacity style={styles.button} onPress={() => ferramentasSat("BloquearSat")}>
                        <Text style={styles.textButton}>BLOQUEAR SAT</Text>
                    </TouchableOpacity>

                    <TouchableOpacity style={styles.button} onPress={() => ferramentasSat("DesbloquearSat")}>
                        <Text style={styles.textButton}>DESBLOQUEAR SAT</Text>
                    </TouchableOpacity>

                    <TouchableOpacity style={styles.button} onPress={() => ferramentasSat("ExtrairLog")}>
                        <Text style={styles.textButton}>EXTRAIR LOG</Text>
                    </TouchableOpacity>

                    <TouchableOpacity style={styles.button} onPress={() => ferramentasSat("AtualizarSoftware")}>
                        <Text style={styles.textButton}>ATUALIZAR SOFTWARE</Text>
                    </TouchableOpacity>

                    <TouchableOpacity style={styles.button} onPress={() => ferramentasSat("Versao")}>
                        <Text style={styles.textButton}>VERIFICAR VERSÃO</Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    button: {
        width: 300,
        height: 40,
        backgroundColor: '#c7c7c7',
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 10,
        borderRadius: 5
    },

    container: {
        flex: 1,
        justifyContent: 'flex-start',
        alignItems: 'center'
    },
    containerButtons: {
        width: '35%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingVertical: 20,
    },
    containerTitleAndInput:{
        width: 500,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        marginVertical: 10,
    },
    containerOptionsAndButtons: {
        height: 400,
        width: '35%',
        alignItems: 'center',
        justifyContent: 'center'
    },

    inputText: {
        width: '100%',
        borderBottomColor: 'gray',
        borderBottomWidth: 1,
        fontSize: 18,
        color: 'black',
        textAlignVertical: 'bottom',
        height: 30,
        padding: 0,
    },

    textButton: {
        fontSize: 14,
        fontWeight: 'bold'
    },
    textOptions:{
        fontSize: 18,
        fontWeight: 'bold',
        color: 'black',
        marginTop: 10,
        alignSelf: 'flex-start'
    }
});

export default ferramentasSat;

File: Flutter/Flutter/lib/pages/sat_pages/ferramentasSat.dart
import 'dart:math';

import 'package:flutter/material.dart';
import 'package:flutter_gBot/Configs/config_sat/retornoSat.dart';
import 'package:flutter_gBot/services/operacaoSat.dart';
import 'package:flutter_gBot/util/common_code.dart';
import 'package:flutter_gBot/util/globalValues.dart';
import 'package:flutter_gBot/widgets/widgetsgertec.dart';

class PageFerramentaSat extends StatefulWidget {
  @override
  _PageFerramentaSatState createState() => _PageFerramentaSatState();
}

class _PageFerramentaSatState extends State<PageFerramentaSat> {
  // Inicializa o código de ativação com um valor global, para o usuario não precisar ficar digitando
  final codigoAtivacao = TextEditingController(text: GlobalValues.codAtivarSat); // Codigo de ativação do Sat

  CommonGertec commonGertec = new CommonGertec(); //* Classe que possui partes de código comuns em diversas telas

  // Função para validar os valores digitos pelo usuario e realizar a ativação do SAT
  void ferramentasSat(String funcao) async {
    GlobalValues.codAtivarSat = codigoAtivacao.text; // Salva  o código de ativação para o usuario não precisar ficar digitando em todas as telas
    if (funcao == "Versao" || commonGertec.isCodigoValido(codigoAtivacao.text)) {
      //* Chama a função Invocar Operação Sat, que recebe como parâmetro a "operação invocada" e um "Map com as chaves e seus respectivos valores".
      RetornoSat retornoSat = await OperacaoSat.invocarOperacaoSat(
          // Passa como parâmetro um Map de argumentos de valores que deseja enviar
          args: {'funcao': funcao, 'random': Random().nextInt(999999), 'codigoAtivar': codigoAtivacao.text.toString()});

      //? Caso deseje visualizar o retorno completo basta chamar [retornoSat.getResultadoCompleto]

      // O retorno da operação Versão não é tratado, pois retorna somente uma string
      if (funcao == "Versao") {
        WidgetsGertec.dialogo(retornoSat.getResultadoCompleto, context: context);
      } else {
        //* Está função [OperacaoSat.formataRetornoSat] recebe como parâmetro a operação realizada e um objeto do tipo RetornoSat
        //* Retorna uma String com os valores obtidos do retorno da Operação já formatados e prontos para serem exibidos na tela
        // Recomenda-se acessar a função e entender como ela funciona
        String retornoFormatado = OperacaoSat.formataRetornoSat(retornoSat: retornoSat);
        WidgetsGertec.dialogo(retornoFormatado, context: context);
      }
    } else {
      WidgetsGertec.dialogo("Código de Ativação deve ter entre 8 a 32 caracteres!", context: context);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: Container(
          color: Colors.white,
          padding: EdgeInsets.only(top: 30),
          height: MediaQuery.of(context).size.height,
          width: MediaQuery.of(context).size.width,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisSize: MainAxisSize.min,
            children: <Widget>[
              SizedBox(width: 30),
              WidgetsGertec.formField(codigoAtivacao, "Código de Ativação SAT:", iWidht: 400),
              WidgetsGertec.buttonStandard("BLOQUEAR SAT", callback: () => ferramentasSat("BloquearSat")),
              WidgetsGertec.buttonStandard("DESBLOQUEAR SAT", callback: () => ferramentasSat("DesbloquearSat")),
              WidgetsGertec.buttonStandard("EXTRAIR LOG", callback: () => ferramentasSat("ExtrairLog")),
              WidgetsGertec.buttonStandard("ATUALIZAR SOFTWARE", callback: () => ferramentasSat("AtualizarSoftware")),
              WidgetsGertec.buttonStandard("VERIFICAR VERSÃO", callback: () => ferramentasSat("Versao")),
            ],
          ),
        ),
      ),
    );
  }
}
